### PR TITLE
feat: add get incident by key endpoint to C8 API

### PR DESCRIPTION
**Pull request by ralfpuchert**
## Description

* Adjust fields according to C8 API definition
* Add `GET /v2/indicents/{incidentKey}` endpoint
* Adjust tests
* Add `description` documentation in `rest-api.yaml`

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #

### CODE REVIEW

**Review by pepopowitz — APPROVED** [review]
👍 The spec changes look great! ❤️

**Review by nathansandi — APPROVED** [review]
LGTM

File: zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/IncidentQueryController.java [hunk 1]
```diff
@@ -43,6 +51,31 @@ public ResponseEntity<IncidentSearchQueryResponse> searchIncidents(
         .fold(RestErrorMapper::mapProblemToResponse, this::search);
   }
 
+  @GetMapping(
+      path = "/{incidentKey}",
+      produces = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_PROBLEM_JSON_VALUE})
+  public ResponseEntity<IncidentEntity> getByKey(
```
[the remark this answers is not in the document]
Comment (ralfpuchert) [reply zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/IncidentQueryController.java]: Yes, thanks. I have recognized it later too.

File: zeebe/gateway-protocol/src/main/proto/rest-api.yaml [hunk 2]
```diff
@@ -1513,7 +1513,56 @@ paths:
...
+    get:
+      tags:
+        - Incident
+      summary: |
+        Returns incident as JSON.
+        :::note
+        This endpoint is an alpha feature and not enabled on Camunda clusters out of the box.
+        See the [Camunda 8 REST API overview](/apis-tools/camunda-api-rest/camunda-api-rest-overview.md#query-api)
+        for further details.
+        :::
```
Comment (tmetzke) [inline zeebe/gateway-protocol/src/main/proto/rest-api.yaml:1526]: 🔧 I think we overlooked this here. The summary should be a short text. This is rather the `description` of the path.

Comment (pepopowitz) [reply zeebe/gateway-protocol/src/main/proto/rest-api.yaml:1526]: Good catch, @tmetzke! Looks to me like you already fixed it in https://github.com/camunda/camunda/pull/22755. Thank you!

**Pull request merged**